### PR TITLE
Add md5 hashing of windows

### PR DESCRIPTION
**Pull request by xssChauhan**
## Description
1. Added Endpoint on server as `/win_hash` to generate the hash of a window's contents
2. Added function to visdom.Visdom for getting hash of a window
3. Added function to browser client for generating hash of a window
4. Added function to browser client for fetching hash of a window from server

## How Has This Been Tested?
I checked the hash generated by the both the server and the browser if it was same for a given window, when the contents were same.

For `Visdom.win_hash` I checked the cases when a window is present and window is absent.
In case of a window present, a hash is returned, otherwise `false` is returned.

For the browser client implementation i tested the function that fetches hash for the following:
1. Making request with the correct parameters
2. Making request at the correct base url
3. Receiving the correct hash

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] For JavaScript changes, I have re-generated the minified JavaScript code.

### CODE REVIEW

File: py/visdom/server.py [hunk 1]
```diff
@@ -155,7 +155,8 @@ def __init__(self, port=DEFAULT_PORT, env_path=DEFAULT_ENV_PATH,
             (r"/win_exists", ExistsHandler, {'app': self}),
             (r"/win_data", DataHandler, {'app': self}),
             (r"/delete_env", DeleteEnvHandler, {'app': self}),
-            (r"/(.*)", IndexHandler, {'app': self}),
+            (r"/win_hash", HashHandler, {'app': self}),
+            (r"/(.*)", IndexHandler, {'app': self})
```
Comment (JackUrb) [inline py/visdom/server.py]: nit: retain comma

File: js/main.js [hunk 2]
```diff
@@ -746,6 +748,25 @@ class App extends React.Component {
     this.setState({cols: cols, width: width}, () => {this.relayout()});
   }
 
+  generateWindowHash = (windowId) => {
+    let windowContent = this.state.panes[windowId];
+
+    //Convert JSON data to string with a space of 2. This detail is important.
```
Comment (JackUrb) [inline js/main.js]: Please note why this detail is important (keeping the js hash consistent with the python hash if I'm guessing correctly).

File: py/visdom/server.py (hunk 1 above)
Comment (xssChauhan) [reply py/visdom/server.py]: My bad. Would fix it right away.

File: js/main.js (hunk 2 above)
Comment (xssChauhan) [reply js/main.js]: Yup. This is to ensure that the JSON is converted to identical strings in both the implementations.

**Comment by xssChauhan** [comment]
@JackUrb I have updated the requested changes

**Comment by xssChauhan** [comment]
@JackUrb Any updates with this PR?

**Comment by JackUrb** [comment]
Hi @xssChauhan - sorry about the delay here! Was off on vacation. Will be pulling this later today

**Comment by xssChauhan** [comment]
@JackUrb Thank you for landing this PR. Is there any other help that I can offer with this core improvement?

**Comment by JackUrb** [comment]
Of course - I'll put more context on next steps in the main issue.

**Pull request closed without merging**